File: scripts/go/generate_signature.go
package main

import (
	"crypto"
	"crypto/rsa"
	"crypto/sha256"
	"crypto/x509"
	"encoding/base64"
	"encoding/pem"
	"os"
)

func main() {
	// Get the private key and signature data from environment variables
	privateKeyPEM := os.Getenv("PRIVATE_KEY")
	signatureData := os.Getenv("SIGNATURE_DATA")

	// Decode the PEM block
	block, _ := pem.Decode([]byte(privateKeyPEM))
	if block == nil {
		panic("failed to parse PEM block containing the private key")
	}

	// Parse the private key
	parsedKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
	if err != nil {
		panic(err)
	}

	// Type assert to RSA private key
	privateKey, ok := parsedKey.(*rsa.PrivateKey)
	if !ok {
		panic("key is not an RSA private key")
	}

	// Hash the data
	hashed := sha256.Sum256([]byte(signatureData))

	// Sign the data
	signature, err := rsa.SignPKCS1v15(nil, privateKey, crypto.SHA256, hashed[:])
	if err != nil {
		panic(err)
	}

	// Output the base64 encoded signature
	print(base64.StdEncoding.EncodeToString(signature))
} 

File: scripts/typescript/generate_signature.ts
import * as crypto from 'crypto';

const privateKey = process.env.PRIVATE_KEY;
const signatureData = process.env.SIGNATURE_DATA;

if (!privateKey || !signatureData) {
    console.error('Error: PRIVATE_KEY and SIGNATURE_DATA environment variables are required');
    process.exit(1);
}

try {
    const sign = crypto.createSign('SHA256');
    sign.update(signatureData);
    const signature = sign.sign(privateKey, 'base64');
    console.log(signature);
} catch (error) {
    console.error('Error generating signature:', error);
    process.exit(1);
} 